Add option for specifying local packages group name

Lommelun opened the pull request:
### Short description 📝

This PR introduces an `Project.Options` property `localPackagesGroupName` that can be set to specify a top-level group name for local packages.

### How to test the changes locally 🧐

Expectation: 
Fixture `ios_app_with_local_swift_package` should still work as expected, where packages are listed top-level (under the project).

Expectation:
Adding `.options(localPackagesGroupName: "Any String")` to the Project manifest for the same fixture `ios_app_with_local_swift_package` and regenerating the project should add the package to a top-level group (under the project) named what was specified in the option.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

pepicrft reviewed (commented):
@Lommelun before merging this I'd love to understand the motivation behind it. Is the `Packages` group an Xcode convention? Is it more of your team's preference?

Lommelun commented:
@pepicrft When the packages are listed top-level it can fill up the project navigator quite a bit depending on the amount of local packages. External package dependencies can be collapsed, so it makes sense to opt-in grouping local packages as well. 

The name is specified when setting the option, we use `Packages` in my team; Hence using that term for the screenshot. If the option is left unspecified it will behave as before–listing local packages ungrouped at top-level under the project.

pepicrft commented:
> @pepicrft When the packages are listed top-level it can fill up the project navigator quite a bit depending on the amount of local packages. External package dependencies can be collapsed, so it makes sense to opt-in grouping local packages as well.

> The name is specified when setting the option, we use Packages in my team; Hence using that term for the screenshot. If the option is left unspecified it will behave as before–listing local packages ungrouped at top-level under the project.

Thanks for the context @Lommelun. I think it's a very sensible default, so what if we do it by default and let users prove it's not a sensible default? We generally lean on the side of "convention over configuration" and allow configurability when several users ask for it. I feel this is one of those features that users don't want to configure.

Lommelun commented:
Correct me if I misunderstand you, but from what I understand you would rather that we just use a bool to group them (and use the name `Packages`) rather than specify the group name? OR do this grouping without any option, by default?

For me either works, and I do believe it is a sensible default.

danyf90 commented:
> @pepicrft Correct me if I misunderstand you, but from what I understand you would rather that we just use a bool to group them (and use the name `Packages`) rather than specify the group name? OR do this grouping without any option, by default?
> 
> For me either works, and I do believe it is a sensible default.

Yep, I think the idea is to group them inside Packages by default, without making it customizable for now

Lommelun commented:
I've made the change; now grouping local packages by default into a `Packages` group and removed the option to set it manually. Updated tests accordingly.

Lommelun commented:
I can update the PR description and title to better fit if the updated code works for you.

danyf90 commented on Tests/TuistGeneratorTests/Generator/ProjectDescriptorGeneratorTests.swift:
```diff
@@ -335,6 +335,29 @@ final class ProjectDescriptorGeneratorTests: TuistUnitTestCase {
+        // Given
+        let project = Project.test(
+            packages: [
+                .local(path: try AbsolutePath(validating: "/Packages/LocalPackageA")),
+                .local(path: try AbsolutePath(validating: "/Packages/LocalPackageB")),
```
I would maybe use a different folder name rather than `Packages` to make clear that's not what defines the group name

Lommelun replied:
Renamed it to `/LocalPackages/...`, hope that is distinct enough 😄 We can of course rename if not.
